components/zip321/src/lib.rs

```diff
@@ -123,17 +83,62 @@ pub struct Payment {
     ///
     /// [`recipient_address`]: #structfield.recipient_address
     memo: Option<MemoBytes>,
-    /// A human-readable label for this payment within the larger structure
-    /// of the transaction request.
-    label: Option<String>,
-    /// A human-readable message to be displayed to the user describing the
-    /// purpose of this payment.
-    message: Option<String>,
-    /// A list of other arbitrary key/value pairs associated with this payment.
-    other_params: Vec<(String, String)>,
+
+    inner: zip321_parse::Payment,
+}
```

Comment:
`zip321::Payment` is essentially a pointer to a parsed `zip321_parse::Payment`, and derefs as such. It contains the validated domain specific types for address, amount and memo on top.